Sources/DarkModeCore/DMDynamicColor.m

```diff
@@ -19,25 +18,48 @@ @implementation DMDynamicColorProxy
 
 // TODO: We need a more generic initializer.
 - (instancetype)initWithLightColor:(UIColor *)lightColor darkColor:(UIColor *)darkColor {
-  self.lightColor = lightColor;
-  self.darkColor = darkColor;
+  return [self initWithDynamicProvider:^(DMTraitCollection *traitCollection){
+    return traitCollection.userInterfaceStyle == DMUserInterfaceStyleDark ? darkColor : lightColor;
+  }];
+}
 
+- (instancetype)initWithDynamicProvider:(UIColor * (^)(DMTraitCollection *traitCollection))dynamicProvider {
+  self.dynamicProvider = dynamicProvider;
```

Comment:
This potentially increases risk of retain cycle. Right?

Reply:
I believe this is meant to be retained here

UIColor's implementation can also result in similar behavior, the color owns the block and the block can capture objects used in the block. I think it is up to developers to sort out this issue. 

<img width="1023" alt="image" src="https://user-images.githubusercontent.com/17924287/84251096-75734780-ab3f-11ea-8b60-5ea121bd36e6.png">